### ISSUE TITLE
./mwaa-local-env build-image fails

### ISSUE THREAD

**Issue by BernardOrzechowski** [aws/aws-mwaa-local-runner#7]
Hi,

on my ubuntu 20.04 (running on vmware) I just tried to build the image and it failed (Cannot find a valid baseurl for repo: amzn2-core/2/x86_64..). I can try to fix it locally, was just wondering if this is something that you also see : 

./mwaa-local-env build-image

Sending build context to Docker daemon  19.95kB
Step 1/25 : FROM amazonlinux
latest: Pulling from library/amazonlinux
3c2c91c7c431: Pull complete 
Digest: sha256:06b9e2433e4e563e1d75bc8c71d32b76dc49a2841e9253746eefc8ca40b80b5e
Status: Downloaded newer image for amazonlinux:latest
 ---> 53ef897d731f
Step 2/25 : LABEL maintainer="amazon"
 ---> Running in 3465e285216b
Removing intermediate container 3465e285216b
 ---> a151fe0f7653
Step 3/25 : ARG AIRFLOW_VERSION=1.10.12
 ---> Running in 05afb75b71b6
Removing intermediate container 05afb75b71b6
 ---> 5b4993d58b04
Step 4/25 : ARG AIRFLOW_USER_HOME=/usr/local/airflow
 ---> Running in 5d5cb8815694
Removing intermediate container 5d5cb8815694
 ---> 1ad36c0f6675
Step 5/25 : ARG AIRFLOW_DEPS=""
 ---> Running in 87985002a9f0
Removing intermediate container 87985002a9f0
 ---> 57d6f50d7b58
Step 6/25 : ARG PYTHON_DEPS=""
 ---> Running in 5dab1621c728
Removing intermediate container 5dab1621c728
 ---> 2a7007526538
Step 7/25 : ARG SYSTEM_DEPS=""
 ---> Running in 5f38c7919606
Removing intermediate container 5f38c7919606
 ---> 586c15a57069
Step 8/25 : ARG INDEX_URL=""
 ---> Running in 7c55422b3812
Removing intermediate container 7c55422b3812
 ---> 2b988b48da95
Step 9/25 : ENV AIRFLOW_HOME=${AIRFLOW_USER_HOME}
 ---> Running in 705cfceb6732
Removing intermediate container 705cfceb6732
 ---> 34dbf4e14682
Step 10/25 : COPY script/bootstrap.sh /bootstrap.sh
 ---> bc966836b5d0
Step 11/25 : COPY script/systemlibs.sh /systemlibs.sh
 ---> 14e5c9ae9c32
Step 12/25 : COPY config/constraints.txt /constraints.txt
 ---> e25943d55d4f
Step 13/25 : COPY config/requirements.txt /requirements.txt
 ---> cf80151bae6d
Step 14/25 : RUN chmod u+x /systemlibs.sh && /systemlibs.sh
 ---> Running in e71c062050f3
Loaded plugins: ovl, priorities

 One of the configured repositories failed (Unknown),
 and yum doesn't have enough cached data to continue. At this point the only
 safe thing yum can do is fail. There are a few ways to work "fix" this:

     1. Contact the upstream for the repository and get them to fix the problem.

     2. Reconfigure the baseurl/etc. for the repository, to point to a working
        upstream. This is most often useful if you are using a newer
        distribution release than is supported by the repository (and the
        packages for the previous distribution release still work).

     3. Run the command with the repository temporarily disabled
            yum --disablerepo=<repoid> ...

     4. Disable the repository permanently, so yum won't use it by default. Yum
        will then just ignore the repository until you permanently enable it
        again or use --enablerepo for temporary usage:

            yum-config-manager --disable <repoid>
        or
            subscription-manager repos --disable=<repoid>

     5. Configure the failing repository to be skipped, if it is unavailable.
        Note that yum will try to contact the repo. when it runs most commands,
        so will have to try and fail each time (and thus. yum will be be much
        slower). If it is a very temporary problem though, this is often a nice
        compromise:

            yum-config-manager --save --setopt=<repoid>.skip_if_unavailable=true

Cannot find a valid baseurl for repo: amzn2-core/2/x86_64
Could not retrieve mirrorlist http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list error was
12: Timeout on http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list: (28, 'Resolving timed out after 5001 milliseconds')
The command '/bin/sh -c chmod u+x /systemlibs.sh && /systemlibs.sh' returned a non-zero code: 1

**Comment by larrykooper** [comment]
I have the same problem

**Comment by larrykooper** [comment]
Problem has cleared up

**Comment by subashcanapathy** [comment]
Closing issue as this was a transient problem on the repo URLs

**Comment by edwardkck** [comment]
Hi @subashcanapathy, I've been running into this issue today with the exact error message above when trying to build the image. 

Can you please elaborate more on the transient problem on the repo URLs?

**Comment by anupash147** [comment]
not solved.. AWS has so many problem, I think my organization is doing the right thing moving to GCP

**Comment by rico-p** [comment]
Hello, 

I am using v2.4.3 on Ubuntu 20.04.5 LTS and the problem still occurs. 

```
./mwaa-local-env build-image
...
Cannot find a valid baseurl for repo: amzn2-core/2/x86_64
Could not retrieve mirrorlist http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list error was
12: Timeout on http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list: (28, 'Resolving timed out after 5000 milliseconds')
The command '/bin/sh -c chmod u+x /systemlibs.sh && /systemlibs.sh' returned a non-zero code: 1
```

Update: it looks like this happens only in case (the company) VPN is enabled. After switching VPN off it works.  
It is just tricky when the VPN is needed for work.

**Comment by davetbo-amzn** [comment]
This is happening to me on my laptop. It looks like the permissions on the CDN are mixed up. It says it can't get the mirror.list from:
http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list

If I click that link it downloads mirror.list and there's one entry in the list:
https://cdn.amazonlinux.com/2/core/2.0/x86_64/b740003e1fd97671e6529530def48ca903a93b24bd100ab0f23f09144b222124

If I click that, I get access denied:
```
<Error>
  <Code>AccessDenied</Code>
  <Message>Access Denied</Message>
  <RequestId>NE7V2WHK6JG5999Z</RequestId>
  <HostId>
x8dLJXRsK106CaHEBwKhjuOzXhfFpBodneRAhzlytkjMkMfk7S/NfeOAi8qDn20Sa70W5JOBGUo=
  </HostId>
</Error>
```

**Comment by yzhishko** [comment]
Same issue here

**Comment by ok512** [comment]
I've created new issue [here](https://github.com/aws/aws-mwaa-local-runner/issues/285), because this on is closed/resolved already.

**Comment by ecc256** [comment]
Running on AWS workspace Ubuntu 22.04, got same result as https://github.com/aws/aws-mwaa-local-runner/issues/7#issuecomment-1475304659

**curl http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list**
`https://cdn.amazonlinux.com/2/core/2.0/x86_64/95051b33cb06275a2be90f45ee8fbd12a95dc2afe7ba94bff90e197f4d6a9403`

**curl https://cdn.amazonlinux.com/2/core/2.0/x86_64/95051b33cb06275a2be90f45ee8fbd12a95dc2afe7ba94bff90e197f4d6a9403**
```
<Error>
  <Code>AccessDenied</Code>
  <Message>Access Denied</Message>
  <RequestId>NE7V2WHK6JG5999Z</RequestId>
  <HostId>
x8dLJXRsK106CaHEBwKhjuOzXhfFpBodneRAhzlytkjMkMfk7S/NfeOAi8qDn20Sa70W5JOBGUo=
  </HostId>
</Error>
```

**Comment by LuciaQX** [comment]
Any solution for this? I have the same error:
yum install -y git
Loaded plugins: ovl
Cannot find a valid baseurl for repo: amzn2-core/2/x86_64
Could not retrieve mirrorlist http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list error was
Timeout on http://amazonlinux.default.amazonaws.com/2/core/latest/x86_64/mirror.list: (28, 'Resolving timed out after 5000 milliseconds')  2023-9-21

**Comment by prashanth-p** [comment]
I identified the solution for this issue.

This is most probably because we executed:
git clone https://github.com/aws/aws-mwaa-local-runner.git in windows cmd

and we are then running ./mwaa-local-env build-image in ubuntu wsl

We need to run dos2unix on all the files in folder ./mwaa-local-env.

The easier fix is to delete the entire folder mwaa-local-env. Then execute 
$ git clone https://github.com/aws/aws-mwaa-local-runner.git

in ubuntu wsl.

Hope this helps
